corehq/apps/app_manager/models.py

```diff
@@ -6834,6 +6837,16 @@ def save(self, *args, **kwargs):
         super(GlobalAppConfig, self).save(*args, **kwargs)
 
 
+class LatestEnabledBuildProfiles(models.Model):
+    app_id = models.CharField(max_length=255)
+    build_profile_id = models.CharField(max_length=255)
+    version = models.IntegerField()
+    build_id = models.CharField(max_length=255)
+
+    def expire_cache(self, domain):
+        get_latest_enabled_build_for_profile.clear(domain, self.build_profile_id)
+        get_enabled_build_profiles_for_version.clear(self.build_id, self.version)
+
```

Comment:
@esoergel i actually want to reconsider the caching done.
1. `get_latest_enabled_build_for_profile`: This is probably the one useful caching. Its loads the app doc and cache's it. In [here](https://github.com/dimagi/commcare-hq/pull/22736/files#diff-2db3dad74bda5ec40f59c83bd340b578R73) we were originally loading an app doc anyway so this was not an extra doc fetch. Caching app doc seems dangerous and serving a stale doc can lead to bad situations. I would like to cache just result of just the [sql query](https://github.com/dimagi/commcare-hq/pull/22736/files#diff-8a78f7edae07ff768accdb85fbaace47R698)
2. `get_enabled_build_profiles_for_version`: i would like to remove this. This method is only used when one is viewing source files to update enabled app profiles, which is going to be very rare and not worth caching.
What do you think?

Reply:
> Caching app doc seems dangerous
I don't agree with that if we're talking specifically about builds (as opposed to the current version of the app) - we use `get_app_cached` in restores normally.  That said, fetching a single document by ID also seems fine.

I agree that it doesn't seem necessary to cache `get_enabled_build_profiles_for_version`.  Was actually thinking that when reviewing this the other day.